drivers/pwm/pwm_mchp_xec.c

```diff
@@ -386,6 +434,12 @@ static int pwm_xec_init(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+#define XEC_PWM_PM_DEV_DEFINE(i) PM_DEVICE_DT_INST_DEFINE(i, pwm_xec_pm_action);
+#else
+#define XEC_PWM_PM_DEV_DEFINE(i) /* Not used */
+#endif
+
```

Comment:
`PM_DEVICE_DT_INST_DEFINE()` is already a no-op if `CONFIG_PM_DEVICE` is not defined.

Reply:
But we have wrapped "pwm_xec_pm_action" definition with "CONFIG_PM_DEFINE". 
So assume if we did not include CONFIG_PM_DEFINE=y in prj.conf, we get compile time error on PM_DEVICE_DT_INST_DEFINE().

Reply:
@henrikbrixandersen any suggestion regarding the above?

Reply:
It's fine to wrap `pwm_xec_pm_action()` in `#ifdef CONFIG_PM_DEVICE`, but the call to `PM_DEVICE_DT_INST_DEFINE()` does not need to be wrapped.

Reply:
Sorry, let me explain the issue i am facing.

We have,
**#ifdef CONFIG_PM_DEVICE**
static int pwm_xec_pm_action(const struct device *dev, enum pm_device_action action)
{
....
}
**#endif**

**#ifdef CONFIG_PM_DEVICE**
#define XEC_PWM_PM_DEV_DEFINE(i) PM_DEVICE_DT_INST_DEFINE(i, pwm_xec_pm_action);
**#else**
#define XEC_PWM_PM_DEV_DEFINE(i) /* Not used */
**#endif**

Let us remove the wrapper for "PM_DEVICE_DT_INST_DEFINE", so the code looks below.
#define XEC_PWM_PM_DEV_DEFINE(i) PM_DEVICE_DT_INST_DEFINE(i, pwm_xec_pm_action);

I compile the code with PM feature, no issues. But if i try compile code without PM feature i am getting error on 
"#define XEC_PWM_PM_DEV_DEFINE(i) PM_DEVICE_DT_INST_DEFINE(i, pwm_xec_pm_action);" line.

Because  "static int pwm_xec_pm_action(const struct device *dev, enum pm_device_action action)" definition is inside the #ifdef CONFIG_PM_DEVICE.

Shall i propose below?
static int pwm_xec_pm_action(const struct device *dev, enum pm_device_action action)
{
int ret = 0;
**#ifdef CONFIG_PM_DEVICE**
....
**#endif**
return ret;
}

Reply:
Please take a look on how other drivers handle this, one example: https://github.com/zephyrproject-rtos/zephyr/blob/6719caf05e0fc43166a959fdefa5375789dc4320/drivers/sensor/lm77/lm77.c#L414

Reply:
FYR, below is the build error if i did not added "CONFIG_PM_DEVICE=y" in prj.conf
![image](https://github.com/zephyrproject-rtos/zephyr/assets/123922411/0b3afe5a-3020-485e-b0bd-f37d03703cd5)

I am sure, they should have PM enabled in their proj.conf. So it got compiled.
Let me compile lm77.c from my side once with out PM, and update you.

Reply:
Thanks @henrikbrixandersen.
Able to compile lm77.c successfully (without PM).

Below change i observed in pwm driver when i did a compare against lm77.c file. 
 "#include <zephyr/pm/device.h>" wrapped with "CONFIG_PM_DEVICE".
I removed the wrapper and able to compile.